### DISCUSSION TITLE
Tinygo

### DISCUSSION THREAD

**Discussion by EduardShaid** [vlang/v#12826] [Ideas]
Can we expect an analogue of the project for V, like https://github.com/tinygo-org/tinygo?

**Comment by JalonSolov** [comment]
In V, we call that the `-native` option.  :-)

Seriously, one of the main reasons tinygo exists is because the executables Go normally produces are huge.  V executables are already usually smaller that equivalent Go code, and with `-native` they are often a tiny fraction of the size.

`-native` is still WIP, so you can't use it for everything, yet, but as an example...

$ **v examples/hello_world.v** # with tcc default
$ ls -l examples/hello_world
-rwxr-xr-x 1 jamie jamie **574816** Dec 13 13:44 examples/hello_world
$ **v -skip-unused examples/hello_world.v** # tcc, with some unused stuff stripped
$ ls -l examples/hello_world
-rwxr-xr-x 1 jamie jamie **291496** Dec 13 13:45 examples/hello_world
$ **v -skip-unused -prod examples/hello_world.v** # some unused stuff stripped, then compiled with gcc with -O3
$ ls -l examples/hello_world
-rwxr-xr-x 1 jamie jamie **84968** Dec 13 13:45 examples/hello_world
$ **v -native examples/hello_world.v** # the option I've been talking about...
$ ls -l examples/hello_world
-rwxrwxr-x 1 jamie jamie **186** Dec 13 13:45 examples/hello_world
$ 

All sizes above in _bytes_.

**Reply by jorgeluismireles** [reply]
Where can I find what **-native** supports and what does not?

**Reply by JalonSolov** [reply]
There is no such list.  You just have to try it and see if it works for your code.  Don't expect a lot from it.

**Reply by jorgeluismireles** [reply]
Thanks. Congratulations to the V language team!
